Add Redis Lock with Test Unitary

silverpark opened the pull request:
Hello,

You can validate the pull request. I have add Redis Lock with Test Unitary and I have used psr-2 for coding style.

Thank you.

Regards.

hrakotobe commented:
Hello

We would like this class to initiate its own Redis connection. This would enable a caller to swap this class with any of the other lock classes, for instance the FileLock class, with minimal effort.

Could you add a constructor taking connection parameters (an array of string for instance) and creating the Redis client object from these parameters ?
Having the lock key also passed to the constructor could be convenient as well.

As an aside, the `LockInterface#getInfo` could return these initialisation parameters.

Regards

silverpark commented:
Hello,

I've updated the constructor to get the connection parameters and lock key with test unitary. I also added the creation of the Redis client object from these parameters.

I think that receive Redis client object in parameters of constructor is better if you add independency injection. Swapping class should not be a problem.

Best regards.

dav-m85 commented:
@hrakotobe I would need this feature too, it is mergeable ?
